spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/GqlDatastoreQuery.java

```diff
@@ -118,7 +117,28 @@ public Object execute(Object[] parameters) {
 		List<T> rawResult = found == null ? Collections.emptyList()
 				: StreamSupport.stream(found.spliterator(), false)
 						.collect(Collectors.toList());
-		return applyProjection(rawResult);
+		Object result;
+		if (isCountQuery()) {
+			result = rawResult.size();
+		}
+		else if (isExistsQuery()) {
+			result = !rawResult.isEmpty();
```

Comment:
`EXISTS` also seems to be supported. So, like in the comment above, I don't think we need to do anything special.

Reply:
This is for datastore!

Reply:
Oops! I thought I was looking at the Spanner one :)